src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -530,3 +543,37 @@ export const performEncryptPostSubmissionActions = (
       })
     })
 }
+
+export const getQuarantinePresignedPostData = async (
+  attachmentSizes: AttachmentSizeMapType[],
+): Promise<
+  Result<
+    AttachmentPresignedPostDataMapType[],
+    CreatePresignedPostError | InvalidFieldIdError
+  >
+> => {
+  const attachmentPresignedData: AttachmentPresignedPostDataMapType[] = []
+  const totalAttachmentSizeLimit = fileSizeLimitBytes(FormResponseMode.Encrypt)
+  let totalAttachmentSize = 0
+  for (const { id, size } of attachmentSizes) {
```

Comment:
ah, so awaiting in a for loop means the promise in each iteration must resolve before the next iteration of the loop proceeds

in this case, `createPresignedPostdataPromise` involved an asynchronous network call. if there are 100 (small) attachments, then this will take 100 round trips sequentially before the `getQuarantinePresignedPostData` function returns :)

Suggestion: try to rewrite this as `Promise.all(attachmentSizes.map(...))`, so that the create presigned calls execute concurrently

Reply:
Thanks! Sorry it took a while, refactored here: [522ed98](https://github.com/opengovsg/FormSG/pull/6685/commits/522ed98d86b954430cd856a05e4de398f080d241)

Reply:
Sorry, refactored again here: [9b0eadb](https://github.com/opengovsg/FormSG/pull/6685/commits/9b0eadb790495774520ea3d3bc50d259bd6e9212)
Realised that we should use `ResultAsync.combine` instead of `Promise.all` for cleaner implementation.